quinn-proto/src/connection/mod.rs

```diff
@@ -1219,28 +1258,31 @@ where
                 && self.spaces[SpaceId::Handshake].crypto.is_none())
     }
 
-    fn set_loss_detection_timer(&mut self) {
-        if let Some((loss_time, _)) = self.earliest_time_and_space(|x| x.loss_time) {
+    fn set_loss_detection_timer(&mut self, now: Instant) {
+        if let Some((loss_time, _)) = self.loss_time_and_space() {
             // Time threshold loss detection.
             self.timers.set(Timer::LossDetection, loss_time);
             return;
         }
 
-        // Don't arm timer if there are no ack-eliciting packets
-        // in flight and the handshake is complete.
+        if self.anti_amplification_blocked() {
```

Comment:
Shouldn't this come before the preceding block?

Reply:
We want to detect packet loss in a timely fashion even if we can't retransmit. Lost packet detection only happens on LossDetection timeout or when we receive an ACK from the client. If the server's first flight is lost and it then becomes blocked on anti-amplification, the client will only send an anti-deadlock packet that contains no ACKs, and the server might not have new data to send, leading to a deadlock unless time loss detection remains effective.

This is transcribed directly from [the draft's pseudocode](https://quicwg.org/base-drafts/draft-ietf-quic-recovery.html#name-setting-the-loss-detection-).

Reply:
Nevermind, I was confused about how time threshold loss detection works--it's actually not used at all unless a later packet in a given space was already acked. That makes it rather difficult for it to arise when the peer's address hasn't yet been validated, but I'm not sure it's impossible...

Reply:
Yeah, I see what I was thinking here doesn't make sense.